database.go

```diff
@@ -63,6 +64,22 @@ func sameNameOnDB(newBook Book) (unique bool) {
 	}
 }
 
+/* Search a book in database based on ID and returns it if succeed. */
+func searchById(id uuid.UUID) (empty bool, storedBook Book) {
```

Comment:
You should return an `error` here,
Wrap it with an `fmt.Errorf` call, add some context data in the message.
Even the `sql.ErrNoRows` can be wrapped.

Reply:
To know exactly what error you returned you can use the `errors.Is` or `error.As`.
https://pkg.go.dev/errors#As
https://pkg.go.dev/errors#Is

Reply:
I think I'm having a misunderstanding here. Inside the searchById function, we have just the methods QueryRow and Scan, that together seems to return only one possible error, that is ErrNoRows. The way I've written the function, the ErrNoRows is not actually wrapped, but it's treated as the 'empty' bool in line 64. Which else error do I have that I could wrap? I'm not seeing it...

Reply:
The one you calling `panic` on.